### PR TITLE
Exercises and advanced traitsui slides

### PR DESCRIPTION
**Pull request by siddhantwahal**

### CODE REVIEW

File: slides/03_traitsui.md [hunk 1]
```diff
@@ -313,9 +351,141 @@ sam = Person(name='Sam', age=29, bff=frodo)
...
+        axes = figure.add_subplot(111)
+        axes.imshow(chelsea())
+        return figure
+
+ImageViewer().edit_traits()
+```
+<!-- #region slideshow={"slide_type": "slide"} -->
+## The ModelView object:
+- We want our science model to be free of UI code
+- But it's still useful for models and views to responde to changes to one
```
Comment (jonathanrocher) [inline slides/03_traitsui.md:426]: `responde` -> `respond`

File: slides/03_traitsui.md [hunk 2]
```diff
@@ -313,9 +351,141 @@ sam = Person(name='Sam', age=29, bff=frodo)
...
+    def _data_default(self):
+        return chelsea()
+
+    def _figure_default(self):
+        figure = Figure()
+        axes = figure.add_subplot(111)
+        axes.imshow(chelsea())
+        return figure
+
+ImageViewer().edit_traits()
```
Comment (jonathanrocher) [inline slides/03_traitsui.md:421]: `configure_traits`?

File: slides/03_traitsui.md [hunk 3]
```diff
@@ -313,9 +351,141 @@ sam = Person(name='Sam', age=29, bff=frodo)
...
+        - What mechanism would we use to hide the `DataFrame` if the directory doesn't have any images
+          and instead show a helpful message? 
+        - Hint: keyword arguments for `Item`
+<!-- #endregion -->
+<!-- #region slideshow={"slide_type": "slide"} -->
+## Solution
+- `stage3.1_first_views/traited_face_detect.py`
+- `stage3.1_first_views/image_folder_view.py`
+<!-- #endregion -->
 ## Toolkit selection
```
Comment (jonathanrocher) [inline slides/03_traitsui.md:489]: I would remove that section.

**Comment by siddhantwahal** [comment]
> Couple of small things and 1 question: I always thought you needed to use `configure_traits` rather than `edit_traits` if running a script, so the GUI event loop is created. Is the code in there supposed to be run inside jupyter?

There's `%gui qt` which embeds the IPython event loop inside qt. Then when you do edit traits, both the UI and the REPL are live so you can push changes from one to the other.

**Pull request merged**